src/mongocxx/test_util/client_helpers.cpp

```diff
@@ -382,7 +575,7 @@ bool matches(types::bson_value::view main,
         array::view main_array = main.get_array().value;
         array::view pattern_array = pattern.get_array().value;
 
-        if (main_array.length() < pattern_array.length()) {
+        if (size(main_array) < size(pattern_array)) {
```

Comment:
We should test the arrays have the same number of elements, not that they are the same length. This caused issues when I originally tried to use this function for the unified test format. I reverted most of the changes to this function after creating my own but left this since it's correct.

Reply:
Yeah. If `length`s are equal that doesn't imply that the `size`s is equal as well. I'm not really sure why we need `length`. Maybe to check to make sure you're under the 16MB limit?

Reply:
I'm not sure. It could have also been a mistake in the original API.